app/src/test/java/com/example/bottomnav/ParseTest.java

```diff
@@ -0,0 +1,171 @@
+                        list.add(contents);
+                    }
+                }
+                else {
+                    if (contents.typeQualifer.equals(key)) {
+                        list.add(contents);
+                    }
+                }
+            }
+            if (list.isEmpty()) {
+                throw new IllegalArgumentException("No contents[s] were found for given key.");
+            }
+            return list;
+        }
+    }
+
+    @Test
+    public void doesNotExist() throws Exception {
+        String code = "const uint16_t CAN_ID;";
+        Parse test = new Parse(code);
+
+        assertEquals(null, test.get("CAN_ID"));
+        assertEquals(null, test.get("ducky"));
+    }
+
+    @Test
+    public void simpleCheck() throws Exception {
+        String code = "const uint16_t CAN_ID = 0x16; \n" +
+                "const uint16_t CAN_ID2 = 0x32; \n" +
+                "const uint16_t CAN_ID3 = 0x64;";
+
+        Parse test = new Parse(code);
+
+        HashMap<String, Contents> repository = test.constRepo;
+
+        assertEquals(true, repository.containsKey("CAN_ID"));
+        assertEquals(true, repository.containsKey("CAN_ID2"));
+        assertEquals(true, repository.containsKey("CAN_ID3"));
+
+        assertEquals("const", test.get("CAN_ID").typeQualifer);
+        assertEquals("const", test.get("CAN_ID2").typeQualifer);
+        assertEquals("const", test.get("CAN_ID3").typeQualifer);
+
+        assertEquals("uint16_t", test.get("CAN_ID").type);
+        assertEquals("uint16_t", test.get("CAN_ID2").type);
+        assertEquals("uint16_t", test.get("CAN_ID3").type);
+
+        assertEquals("0x16", test.get("CAN_ID").value);
+        assertEquals("0x32", test.get("CAN_ID2").value);
+        assertEquals("0x64", test.get("CAN_ID3").value);
+    }
+
+    @Test
+    public void checkdiverse() throws Exception {
+        String code = "const uint16_t CAN_ID1 = 0x16;\n" +
+                "const uint32_t CAN_ID3 = 0x64;\n" +
+                "const uint32_t CAP_ID2 = 0x18;";
+
+        Parse test = new Parse(code);
+
+        HashMap<String, Contents> repository = test.constRepo;
+
+        assertEquals(true, repository.containsKey("CAN_ID1"));
+        assertEquals(true, repository.containsKey("CAP_ID2"));
+        assertEquals(true, repository.containsKey("CAN_ID3"));
+
+        assertEquals("const", test.get("CAN_ID1").typeQualifer);
+        assertEquals("const", test.get("CAP_ID2").typeQualifer);
+        assertEquals("const", test.get("CAN_ID3").typeQualifer);
+
+        assertEquals("uint16_t", test.get("CAN_ID1").type);
+        assertEquals("uint32_t", test.get("CAP_ID2").type);
+        assertEquals("uint32_t", test.get("CAN_ID3").type);
+
+        assertEquals("0x16", test.get("CAN_ID1").value);
+        assertEquals("0x18", test.get("CAP_ID2").value);
+        assertEquals("0x64", test.get("CAN_ID3").value);
+    }
+
+    @Test
+    public void listCheck() throws Exception {
```

Comment:
This seems like it's more testing the `MakeList` utility class? What's `MakeList` for? It seems like its only purpose is to be tested, which is kind of odd?